consensus/integ/integ_test.go

```diff
@@ -0,0 +1,144 @@
+package integ
+
+import (
+	"testing"
+	"time"
+
+	"github.com/NethermindEth/juno/consensus/driver"
+	"github.com/NethermindEth/juno/consensus/tendermint"
+	"github.com/NethermindEth/juno/consensus/types"
+	"github.com/NethermindEth/juno/db/pebble"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type testConfig struct {
+	nodeCount          int
+	silentFaultyNodes  bool
+	buffer             int
+	timeoutBase        time.Duration
+	timeoutStepFactor  time.Duration
+	timeoutRoundFactor time.Duration
+	targetHeight       types.Height
+}
+
+func getTimeoutFn(cfg testConfig) func(types.Step, types.Round) time.Duration {
+	return func(step types.Step, round types.Round) time.Duration {
+		return cfg.timeoutBase + time.Duration(step)*cfg.timeoutStepFactor + time.Duration(round)*cfg.timeoutRoundFactor
+	}
+}
```

Comment:
I was going to write a whole comment here saying that because the fields of `cfg` are copied value, no closure would happen over `cfg` so passing it by value (in `getTimeoutFn`) looking for guarantees of immutability inside the delegate wasn't necessary and we could pass by reference (i.e. `cfg *testConfig`) but I think I am wrong. Passing by value for the `getTimeoutFn` makes sense.